script/validators/source_validator.py

```diff
@@ -219,8 +219,35 @@ def check_namespaces(file_path):
     return status
 
 
+def check_includes(file_path):
+    """Checks whether local includes are done via #include<...>"""
+    with open(file_path, "r") as f:
+        path_pattern = re.compile(r'<(.*?)>')
+        includes = os.listdir('src/include')
+        linenum = 0
+        status = True
+        for line in f:
+            linenum += 1
+            if line.startswith('#include <'):
```

Comment:
Did you know, you can actually test this also in the regex: https://regexr.com/3j66j (with multiline flag)

Reply:
That's quite cool! But then I'd have to dump all the lines in the source at once (using `f.readlines()`), right?

Reply:
You can test every line with the regex and check if a match was found or not. You can also regex the whole file (cmp. [1]), but then you need to output the whole line as you are missing the line numbers. 

[1] https://github.com/schedutron/peloton/blob/cbd505df0b3b3539bc2cf9fa8f4ee15f17c6b299/script/validators/source_validator.py#L172